src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1273,6 +1274,41 @@ bool Literal::is_string_like() const noexcept {
 }
 
 Literal Literal::add(Literal const &other, Node::NodeStorage &node_storage) const noexcept {
+    if ((this->datatype_eq<datatypes::xsd::DateTime>() || this->datatype_eq<datatypes::xsd::DateTimeStamp>() || this->datatype_eq<datatypes::xsd::Date>() || this->datatype_eq<datatypes::xsd::Time>()) &&
```

Comment:
This function is incredibly unreadable. Please add some spaces in appropriate places

Reply:
i want to add here that having comments that describe the cases would make things easier to understand. 
i think currently it is easy to miss some cases.

if it does not affect the performance and it is easy to implement, i think it would be even better if you could move the implementation to separate functions.